### PR TITLE
Making the bot more secure by using the pull_request_target event.

### PR DESCRIPTION
**Pull request by gabrieldemarmiesse**
# Description

Brief Description of the PR:

Even if the token was obfuscated before, technically anyone could run the script to notify the codeowners. This fixes the security issue by making use of the `pull_request_target` event. We'll need to restart the CI job once the token is added.

See for more info: https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request_target

@seanpmorgan  , I sent you the token by email, please add it as a secret github actions, thanks!

## Type of change

- [x] Bug fix
- [ ] New Tutorial
- [ ] Updated or additional documentation
- [ ] Additional Testing
- [ ] New Activation and the changes conform to the [activation contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/activations/README.md#contribution-guidelines)
- [ ] New Callback and the changes conform to the [callback contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/callbacks/README.md#contribution-guidelines)
- [ ] New Image addition and the changes conform to the [image op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/image/README.md#contribution-guidelines)
- [ ] New Layer and the changes conform to the [layer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/layers/README.md#contribution-guidelines)
- [ ] New Loss and the changes conform to the [loss contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/losses/README.md#contribution-guidelines)
- [ ] New Metric and the changes conform to the [metric contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/metrics/README.md#contribution-guidelines)
- [ ] New Optimizer and the changes conform to the [optimizer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/README.md#contribution-guidelines)
- [ ] New RNN Cell and the changes conform to the [rnn contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/rnn/README.md#contribution-guidelines)
- [ ] New Seq2seq addition and the changes conform to the [seq2seq contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/seq2seq/README.md#contribution-guidelines)
- [ ] New Text addition and the changes conform to the [text op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/text/README.md#contribution-guidelines)

# Checklist:

- [x] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#coding-style)
    - [x] By running Black + Flake8
    - [x] By running pre-commit hooks
- [ ] This PR addresses an already submitted issue for TensorFlow Addons
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] This PR contains modifications to C++ custom-ops

# How Has This Been Tested?

We can test it with the ci since I didn't make a fork to make this pull request.

If you're adding a bugfix or new feature please describe the tests that you ran to verify your changes:
*

### CODE REVIEW

**Review by bhack — APPROVED** [review]

File: .github/workflows/notify_codeowners.yml [hunk 1]
```diff
@@ -18,6 +18,7 @@ jobs:
       - name: Drop a message for codeowners
         env:
           PR: ${{ steps.findPr.outputs.pr }}
+          GITHUB_BOT_TOKEN: ${{ secrets.GITHUB_BOT_TOKEN }}
```
Comment (seanpmorgan) [inline .github/workflows/notify_codeowners.yml:21]: Need to modify the secret name to not prefix `GITHUB_`:
https://docs.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets#naming-your-secrets

**Comment by gabrieldemarmiesse** [comment]
By the way, here is the thread concerning the same security issue for the backport: https://github.com/Gaurav0/backport/issues/44

File: .github/workflows/notify_codeowners.yml [hunk 2]
```diff
@@ -18,6 +18,7 @@ jobs:
       - name: Drop a message for codeowners
         env:
           PR: ${{ steps.findPr.outputs.pr }}
+          GITHUB_BOT_TOKEN: ${{ secrets.BOT_TOKEN }}
```
Comment (seanpmorgan) [inline .github/workflows/notify_codeowners.yml:21]:
```suggestion
          BOT_TOKEN: ${{ secrets.BOT_TOKEN }}
```

Or could change the os env variable that is called.

**Comment by bhack** [comment]
Can we merge this?

**Comment by jmckenna** [comment]
Can this be merged?  I'm hitting this same wall.

**Comment by gabrieldemarmiesse** [comment]
Still working on it, I forgot about this PR after a while. I'm trying to make the two bot workflows more secure and reliable. I'll keep you updated

**Comment by gabrieldemarmiesse** [comment]
@seanpmorgan this pull request is ready for review
* Increased security by using the workflow pull_request_target, and thus nobody can get the bot token now, except from the workflow backporting PR, I'll do that later
* Added a cache to avoid looking up the existence of a user multiple times, it takes ~1s per user, so it's actually a great speedup.
* Remove the main argument for the script. It's more logical to check the codeowner file in the current workspace.

File: .github/workflows/notify_codeowners.yml (hunk 2 above)
Comment (gabrieldemarmiesse) [reply .github/workflows/notify_codeowners.yml]: Fixed it

**Review by seanpmorgan — APPROVED** [review]
LGTM Thanks!

**Pull request merged**